AIMSICD/src/main/java/com/secupwn/aimsicd/smsdetection/AdvancedUserActivity.java

```diff
@@ -49,48 +46,32 @@
 
     @InjectView(R.id.spinner)
     private Spinner spinner;
-    private List<AdvanceUserItems> msgItems;
+
+    private Realm realm;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        dbAccess = new AIMSICDDbAdapter(getApplicationContext());
 
-        try {
-            msgItems = dbAccess.getDetectionStrings();
-        } catch (Exception ee) {
-            log.error("Database error", ee);
-            msgItems = new ArrayList<>();
-            AdvanceUserItems advUserItems = new AdvanceUserItems();
-            advUserItems.setDetection_string("NO DATA");
-            advUserItems.setDetection_type("No TYPE");
-            msgItems.add(advUserItems);
-        }
+        realm = Realm.getDefaultInstance();
+
+        RealmResults<SmsDetectionString> msgItems = realm.allObjects(SmsDetectionString.class);
 
-        listViewAdv.setAdapter(new AdvanceUserBaseAdapter(getApplicationContext(), msgItems));
+        listViewAdv.setAdapter(new DetectionStringAdapter(this, msgItems, true));
 
         listViewAdv.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
             @Override
             public boolean onItemLongClick(AdapterView<?> a, View v, int position, long id) {
-                Object o = listViewAdv.getItemAtPosition(position);
-                AdvanceUserItems itemDetails = (AdvanceUserItems) o;
+                SmsDetectionString detectionString = (SmsDetectionString) listViewAdv.getItemAtPosition(position);
 
-                String itemDetail = itemDetails.getDetection_string();
+                String string = detectionString.getDetectionString();
 
-                if (dbAccess.deleteDetectionString(itemDetails.getDetection_string())) {
-                    Toast.makeText(getApplicationContext(),
-                            getString(R.string.deleted) + ": " + itemDetail, Toast.LENGTH_SHORT).show();
-                } else {
-                    Toast.makeText(getApplicationContext(), getString(R.string.failed_to_delete)
-                            + " " + itemDetail, Toast.LENGTH_SHORT).show();
-                }
+                detectionString.removeFromRealm();
```

Comment:
This doesn't looks to be wrapped in a transaction. If it isn't it will crash if called.